uai2017/python/create-invitations.py

```diff
@@ -301,57 +304,68 @@ def get_meta_review_invitation(submissionId, number, authorsGroupId, areachairGr
 
   for submission in submissions:
 
-  	print "Processing submission", submission.number
-
-  	paper_group = client.get_group(id = maskPaperGroup.replace('[PAPER_NUMBER]', str(submission.number)))
-
-  	if paper_group:
-
-  		author_group = client.get_group(id = maskAuthorsGroup.replace('[PAPER_NUMBER]', str(submission.number)))
-
-  		if author_group:
-  			# Post the necessary groups
-  			reviewers_group = openreview.Group(id = paper_group.id + '/Reviewers',
-  				signatures = [CONFERENCE],
-  				writers = [CONFERENCE],
-  				members = [],
-  				readers = [CONFERENCE, COCHAIRS, SPC, paper_group.id + '/Reviewers'],
-  				signatories = [])
-
-  			client.post_group(reviewers_group)
-
-  			reviewers_nonreaders_group = openreview.Group(id = reviewers_group.id + '/NonReaders',
-  				signatures = [CONFERENCE],
-  				writers = [CONFERENCE],
-  				members = [],
-  				readers = [CONFERENCE, COCHAIRS, SPC, reviewers_group.id + '/NonReaders'],
-  				signatories = [])
-
-  			client.post_group(reviewers_nonreaders_group)
-
-  			areachair_group = openreview.Group(id = paper_group.id + '/Area_Chair',
-  				signatures = [CONFERENCE],
-  				writers = [CONFERENCE],
-  				members = [],
-  				readers = [CONFERENCE, COCHAIRS, SPC],
-  				signatories = [CONFERENCE, paper_group.id + '/Area_Chair'])
-
-  			client.post_group(areachair_group)
-
-  			#Post open comment invitation
-  			client.post_invitation(get_open_comment_invitation(submission.id, submission.number, author_group.id))
-  			#Post confidential comment invitation
-  			client.post_invitation(get_confidential_comment_invitation(submission.id, submission.number, author_group.id))
-  			#Post recommend reviewer invitation
-  			client.post_invitation(get_recommend_reviewer_invitation(submission.id, submission.number))
-  			#Post official review invitation
-  			client.post_invitation(get_official_review_invitation(submission.id, submission.number, author_group.id, reviewers_nonreaders_group.id))
-  			#Post meta review invitation
-  			client.post_invitation(get_meta_review_invitation(submission.id, submission.number, author_group.id, areachair_group.id))
-  		else:
-  			print "Author group not found", submission.number
-  	else:
-  		print "Paper group not found ", submission.number
+    print "Processing submission", submission.number
+
+    paper_group = client.get_group(id = maskPaperGroup.replace('[PAPER_NUMBER]', str(submission.number)))
+
+    if paper_group:
+
+      author_group = client.get_group(id = maskAuthorsGroup.replace('[PAPER_NUMBER]', str(submission.number)))
+
+      if author_group:
+        # Check the necessary groups and post them if they don't exist
+        if not client.exists(paper_group.id + '/Reviewers'):
```

Comment:
exists function is doing a get call and then you do an extra get_group. 

I suggest to do a get_group directly and if we get an error do the post:

```
reviewers_group = client.get_group(paper_group.id + '/Reviewers')
if not reviewers_group:
      .. do the post call
```